kms-key/kms-aliases.tf

```diff
@@ -1,4 +1,4 @@
 resource "aws_kms_alias" "a" {
-  name          = "alias/${var.project}/${var.environment}/${var.alias}"
+  name          = replace("alias/${var.project}/${var.environment}/${var.alias}", "/[^a-zA-Z0-9:///_-]+/", "-")
```

Comment:
Good! For old projects where the `replace` **might** actually be triggered, a change in the name will force a replacement.
Since we always refer to the ARNs using references through terraform modules (rather than pasting the literal ARN), and since we apply changes to an entire stack, not just a single resource, this should be fine.

We should still be careful when upgrading old projects though.